torch/distributed/algorithms/ddp_comm_hooks/post_localSGD_hook.py

```diff
@@ -50,6 +51,36 @@ def __init__(
         # Iteration/step in the training loop.
         self.iter = 0
 
+    def __getstate__(self):
+        r"""
+        Returns a ``Dict[str, Any]`` which will be pickled and saved.
+        ``process_group`` and ``subgroup`` are not serializable and excluded from
+        a returned state.
+        """
+        logger.warning(
+            "NOTE: Process group and subgroup are not serializable and excluded from a saved state."
+        )
+        return {
+            slot: getattr(self, slot)
+            for slot in self.__slots__ if slot != "process_group" and slot != "subgroup"
```

Comment:
nit: maybe an isinstance check such as `isinstance(slot, dist.ProcessGroup)` or the like would be more robust?

Reply:
I vaguely recall there being some issues with `dist.ProcessGroup`, where it was not actually defined, which was why we did not always add type annotations for process groups. @rohan-varma do you remember this?

Reply:
This makes sense to me!